sandbox/data_science/app_files/text_to_skills.py

```diff
@@ -0,0 +1,235 @@
+def punct_space(token):
+    """
+    helper function to eliminate tokens
+    that are pure punctuation or whitespace
+    """
+
+    return token.is_punct or token.is_space
+
+def line_review(filename):
+    """
+    SRG: modified for a list
+    generator function to read in reviews from the file
+    and un-escape the original line breaks in the text
+    """
+
+    for review in filename:
+        yield review.replace('\\n', '\n')
+
+def lemmatized_sentence_corpus(filename):
+    """
+    generator function to use spaCy to parse reviews,
+    lemmatize the text, and yield sentences
+    """
+
+    for parsed_review in nlp.pipe(line_review(filename),
+                                  batch_size=10000, n_threads=4):
+
+        for sent in parsed_review.sents:
+            yield u' '.join([token.lemma_ for token in sent
+                             if not punct_space(token)])
+
+######################################################################
+########################### Compute results ##########################
+######################################################################
+
+
+trigram_dictionary = trigram_dictionary_POS
+def vectorize_input(input_doc, bigram_model, trigram_model, trigram_dictionary):
+    """
+    (1) parse input doc with spaCy, apply text pre-proccessing steps,
+    (3) create a bag-of-words representation (4) create an LDA representation
+    """
+
+    # parse the review text with spaCy
+    parsed_doc = nlp(input_doc)
+
+    # lemmatize the text and remove punctuation and whitespace
+    unigram_doc = [token.lemma_ for token in parsed_doc
+                      if not punct_space(token)]
+
+    # apply the first-order and secord-order phrase models
+    bigram_doc = bigram_model[unigram_doc]
+    trigram_doc = trigram_model[bigram_doc]
+
+    # remove any remaining stopwords
+    trigram_review = [term for term in trigram_doc
+                      if not term in stopwords]
+
+    # create a bag-of-words representation
+    doc_bow = trigram_dictionary_POS.doc2bow(trigram_doc)
+
+    # create an LDA representation
+    document_lda = lda[doc_bow]
+    return trigram_review, document_lda
+
+def lda_top_topics(document_lda, topic_names, min_topic_freq=0.05):
+    '''
+    Print a sorted list of the top topics for a given LDA representation
+    '''
+    # sort with the most highly related topics first
+    sorted_doc_lda = sorted(document_lda, key=lambda review_lda: -review_lda[1])
+
+    for topic_number, freq in sorted_doc_lda:
+        if freq < min_topic_freq:
+            break
+
+        # print the most highly related topic names and frequencies
+        print('*'*56)
+        print('{:50} {:.3f}'.format(topic_names[topic_number+1],
+                                round(freq, 3)))
+        print('*'*56)
+        for term, term_freq in lda.show_topic(topic_number, topn=10):
+            print(u'{:20} {:.3f}'.format(term, round(term_freq, 3)))
+        print('\n\n')
```

Comment:
why do all these functions print things instead of returning them?

Reply:
Will use `return`